### PR TITLE
manifest: nrfxlib: nRF Bluetooth Controller -> SoftDevice Controller

### PR DESCRIPTION
**Pull request by rugeGerritsen**
Updates nrfxlib to use the SoftDevice Controller. 

Documentation, API usage, and Kconfig usage have been updated

See CHANGELOG for more details.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

See https://github.com/nrfconnect/sdk-nrfxlib/pull/233

### CODE REVIEW

File: applications/nrf_desktop/doc/ble_qos.rst [hunk 1]
```diff
@@ -98,10 +98,10 @@ Implementation details
 The QoS module uses Zephyr's :ref:`zephyr:settings_api` subsystem to store the configuration in non-volatile memory.
 The channel map is not stored.
 
-Bluetoooth LE controller interaction
+SoftDevice Controller interaction
 ====================================
 
-The module uses CRC information from the Bluetoooth LE controller to adjust the channel map.
+The module uses CRC information from the SoftDevice Controller to adjust the channel map.
```
[the remark this answers is not in the document]
Comment (rugeGerritsen) [reply applications/nrf_desktop/doc/ble_qos.rst:104]: This is the SoftDevice Controller only. The Zephyr BLE controller does not support this feature. @MarekPieta can confirm

[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/doc/ble_qos.rst:104]: the module is working only with our ble_controller; you can set channel map in zephyr but obtaining information about channel quality is not available there

**Comment by rugeGerritsen** [comment]
Pushed to retrigger CI

File: samples/bluetooth/llpm/README.rst [hunk 2]
```diff
@@ -80,7 +80,7 @@ Requirements
   * |nRF52840DK|
   * |nRF52DK|
   * |nRF5340DK|
-  * Other boards running BLE Controller variants that support LLPM (see :ref:`nrfxlib:ble_controller` Proprietary feature support)
+  * Other boards running BLE Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support)
```
Comment (ru-fu) [inline samples/bluetooth/llpm/README.rst:83]:
```suggestion
  * Other boards running Bluetooth LE Controller variants that support LLPM (see :ref:`nrfxlib:softdevice_controller` Proprietary feature support)
```

Comment (ru-fu) [reply samples/bluetooth/llpm/README.rst:83]: There is also a mention of "BLE controller" in line 26 - that could be replaced with "SoftDevice Controller" imo.

File: subsys/bluetooth/controller/hci_driver.c [hunk 3]
```diff
@@ -53,28 +53,28 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
...
+	+ SDC_MEM_SLAVE_LINKS_SHARED)
 
 #define MEMPOOL_SIZE ((CONFIG_BLECTRL_SLAVE_COUNT * SLAVE_MEM_SIZE) + \
 		      (BLECTRL_MASTER_COUNT * MASTER_MEM_SIZE))
 
-static uint8_t ble_controller_mempool[MEMPOOL_SIZE];
+static uint8_t sdc_mempool[MEMPOOL_SIZE];
 
 #if IS_ENABLED(CONFIG_BT_CTLR_ASSERT_HANDLER)
 extern void bt_ctlr_assert_handle(char *file, uint32_t line);
```
[the remark this answers is not in the document]
Comment (rugeGerritsen) [reply subsys/bluetooth/controller/hci_driver.c:80]: Yes, there are still some other things that needs cleanup in this repo. Folder names named bt_ll_nrfxlib etc. Do you want me to fix it in this PR or in a separate one?

[the remark this answers is not in the document]
Comment (joerchan) [reply subsys/bluetooth/controller/hci_driver.c:80]: Separate PR is fine.

**Comment by rugeGerritsen** [comment]
Rebased on master to resolve merge conflict in west.yml

File: samples/bluetooth/llpm/README.rst (hunk 2 above)
Comment (joerchan) [reply samples/bluetooth/llpm/README.rst:83]: BLE in line 20 also.

Comment (rugeGerritsen) [reply samples/bluetooth/llpm/README.rst]: Fixed

File: subsys/bluetooth/controller/Kconfig [hunk 4]
```diff
@@ -41,7 +41,7 @@ config BT_RX_STACK_SIZE
 # when Nordic BLE controller is used.
 config SYSTEM_WORKQUEUE_STACK_SIZE
 	int
-	default 2048 if BT_LL_NRFXLIB
+	default 2048 if BT_LL_SOFTDEVICE
 
 config BLECTRL_SLAVE_COUNT
```
[the remark this answers is not in the document]
Comment (rugeGerritsen) [reply subsys/bluetooth/controller/Kconfig:46]: I would propose to take that in a separate PR

File: samples/bluetooth/llpm/README.rst (hunk 2 above)
Comment (joerchan) [reply samples/bluetooth/llpm/README.rst]: Still BLE in line 20, or do you wish to fix this outside of this PR?

Comment (rugeGerritsen) [reply samples/bluetooth/llpm/README.rst]: We are using BLE in multiple places in this repo. I would suggest to take that in a separate PR

**Comment by rugeGerritsen** [comment]
Rebased on master to retrigger CI

**Comment by rugeGerritsen** [comment]
Pushed to retrigger CI

**Comment by rugeGerritsen** [comment]
Updated the manifest file to point to the master branch

**Pull request merged**